### PR TITLE
bpo-38049: Add command-line interface for the ast module.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue38049

### CODE REVIEW

File: Lib/ast.py [hunk 1]
```diff
@@ -527,3 +527,42 @@ def __new__(cls, *args, **kwargs):
...
+        sys.exit(1)
+
+    # Parse the arguments and options
+    parser = argparse.ArgumentParser(prog='python -m ast')
+    parser.add_argument('infile', type=argparse.FileType(mode='rb'), nargs='?',
+                        default='-',
+                        help='the file to parse; defaults to stdin')
+    parser.add_argument('-m', '--mode', default='exec',
+                        choices=('exec', 'single', 'eval'),
+                        help='the file to parse; defaults to stdin')
```
Comment (ammaraskar) [inline Lib/ast.py]: Looks like you might have forgotten to update the `help` text here.

Comment (serhiy-storchaka) [reply Lib/ast.py]: Thanks!

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -329,6 +329,42 @@ and classes for traversing abstract syntax trees:
...
+.. program:: ast
+
+.. cmdoption:: -h, --help
+
+   Show the help message and exit.
+
+.. cmdoption:: -m <mode>
+               --mode <mode>
+
+   Specify what kind of code must be compiled.
```
Comment (ammaraskar) [inline Doc/library/ast.rst]: This is kinda confusing to someone who is unfamiliar with `compile`'s modes, maybe add a cross-link to the documentation?

File: Doc/library/ast.rst [hunk 3]
```diff
@@ -329,6 +329,43 @@ and classes for traversing abstract syntax trees:
...
+
+.. cmdoption:: -m <mode>
+               --mode <mode>
+
+   Specify what kind of code must be compiled, like the *mode* argument
+   in :func:`parse`.
+
+.. cmdoption:: -a, --include-attributes
+
+   Include attributes such as line numbers and column offsets.
```
Comment (eamanu) [inline Doc/library/ast.rst:373]: Should be great have an example how to pass this params?

Comment (serhiy-storchaka) [reply Doc/library/ast.rst:373]: You just pass them as any other. It is trivial.

**Review by zooba — APPROVED** [review]

**Pull request merged**